tests/ha/multi_pod_restart/test_multi_server_pod_restart.py

```diff
@@ -0,0 +1,163 @@
+        cls.random_time = cls.s3_clean = cls.test_prefix = cls.test_prefix_deg = None
+        cls.s3acc_name = cls.s3acc_email = cls.bucket_name = cls.object_name = cls.node_name = None
+        cls.restore_pod = cls.deployment_backup = cls.deployment_name = cls.restore_method = None
+        cls.qvalue = cls.kvalue = cls.nvalue = None
+        cls.mgnt_ops = ManagementOPs()
+        cls.system_random = secrets.SystemRandom()
+
+        for node in range(cls.num_nodes):
+            cls.host = CMN_CFG["nodes"][node]["hostname"]
+            cls.username.append(CMN_CFG["nodes"][node]["username"])
+            cls.password.append(CMN_CFG["nodes"][node]["password"])
+            if CMN_CFG["nodes"][node]["node_type"] == "master":
+                cls.node_master_list.append(LogicalNode(hostname=cls.host,
+                                                        username=cls.username[node],
+                                                        password=cls.password[node]))
+                cls.hlth_master_list.append(Health(hostname=cls.host,
+                                                   username=cls.username[node],
+                                                   password=cls.password[node]))
+            else:
+                cls.node_worker_list.append(LogicalNode(hostname=cls.host,
+                                                        username=cls.username[node],
+                                                        password=cls.password[node]))
+
+        cls.rest_obj = S3AccountOperations()
+
+    def setup_method(self):
+        """
+        This function will be invoked prior to each test case.
+        """
+        LOGGER.info("STARTED: Setup Operations")
+        self.random_time = int(time.time())
+        self.restore_pod = False
+        self.s3_clean = dict()
+        self.s3acc_name = f"ha_s3acc_{int(perf_counter_ns())}"
+        self.s3acc_email = f"{self.s3acc_name}@seagate.com"
+        LOGGER.info("Precondition: Verify cluster is up and running and all pods are online.")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Precondition: Verified cluster is up and running and all pods are online.")
+        LOGGER.info("Get the value for number pods that can go down for cluster")
+        resp = self.ha_obj.calculate_multi_value(self.csm_obj, len(self.node_worker_list))
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Maximum number of pod that can go down is: %s", resp[1])
+        self.qvalue = resp[1]
+        LOGGER.info("Getting K value for the cluster")
+        resp = self.csm_obj.get_sns_value()
+        if not resp:
+            assert_utils.assert_true(False, "Could not retrieve SNS values of cluster")
+        LOGGER.info("K value for the cluster is: %s", resp[1])
+        self.kvalue = resp[1]
+        self.nvalue = len(self.node_worker_list)
```

Comment:
Can we move qvalue, kvalue and nvalue to setu class, as those won't change everytime, so may be better to keep them in setup class.

Reply:
Hmmm I had similar thought but then thought that in case something fails after making pod down and fails for restoring as well  then it will change the topology so kept it here. Let me know ur thoughts